azext_iot/central/providers/preview/api_token_provider_preview.py

```diff
@@ -0,0 +1,83 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+
+from azext_iot.constants import CENTRAL_ENDPOINT
+from azext_iot.central import services as central_services
+from azext_iot.central.models.enum import Role, ApiVersion
+
+
+logger = get_logger(__name__)
+
+
+class CentralApiTokenProviderPreview:
+    def __init__(self, cmd, app_id: str, token=None):
+        """
+        Provider for API token APIs
+
+        Args:
+            cmd: command passed into az
+            app_id: name of app (used for forming request URL)
+            token: (OPTIONAL) authorization token to fetch API token details from IoTC.
+                MUST INCLUDE type (e.g. 'SharedAccessToken ...', 'Bearer ...')
+                Useful in scenarios where user doesn't own the app
+                therefore AAD token won't work, but a SAS token generated by owner will
+        """
+        self._cmd = cmd
+        self._app_id = app_id
+        self._token = token
+
+    def add_api_token(
```

Comment:
We don't have a strict requirement for type hints yet, but parameter and return type hints go a long way in maintainability.

Reply:
I won't repeat the type hint comment for later places in the PR.

Reply:
fixed